src/Essentials/src/Connectivity/Connectivity.ios.tvos.macos.reachability.cs

```diff
@@ -196,7 +196,7 @@ async void OnChange(NetworkReachabilityFlags flags)
 		{
 			// Add in artifical delay so the connection status has time to change
 			// else it will return true no matter what.
-			await Task.Delay(100);
+			await Task.Delay(400);
```

Comment:
I'm not sure if it's solid enough in all conditions. Instead of a fixed delay, could implement a polling mechanism that checks for actual state changes? Like, check the current network access every 100 ms, maximum 1 second. Trigger the `ReachabilityChanged` event if NetworkAccess changed.

Reply:
> I'm not sure if it's solid enough in all conditions. Instead of a fixed delay, could implement a polling mechanism that checks for actual state changes? Like, check the current network access every 100 ms, maximum 1 second. Trigger the `ReachabilityChanged` event if NetworkAccess changed.

I’ve updated the code as per your suggestion to poll the current network access every 100 milliseconds, for up to 1 second. The ReachabilityChanged event is triggered if a change in NetworkAccess during this interval.